optuna/visualization/matplotlib/_intermediate_values.py

```diff
@@ -1,11 +1,73 @@
 from optuna._experimental import experimental
+from optuna.logging import get_logger
+from optuna.study import Study
+from optuna.trial import TrialState
 from optuna.visualization.matplotlib._matplotlib_imports import _imports
 
 
 if _imports.is_successful():
     from optuna.visualization.matplotlib._matplotlib_imports import Axes
+    from optuna.visualization.matplotlib._matplotlib_imports import plt
+
+_logger = get_logger(__name__)
 
 
 @experimental("2.2.0")
-def plot_intermediate_values() -> Axes:
-    raise NotImplementedError("To be implemented soon.")
+def plot_intermediate_values(study: Study) -> "Axes":
+    """Plot intermediate values of all trials in a study with Matplotlib.
+
+    .. seealso::
+        Please refer to :func:`~optuna.visualization.plot_intermediate_values` for an example.
```

Comment:
Thanks for the swift action! According to the https://github.com/optuna/optuna/pull/1899#discussion_r499364752, we should show the module path. Could you update it?
```suggestion
        Please refer to :func:`optuna.visualization.plot_intermediate_values` for an example.
```

Reply:
Thanks!